mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLTypes.td

```diff
@@ -73,9 +73,10 @@ def SYCL_BufferType : SYCL_Type<"Buffer", "buffer"> {
   let description = [{ Shared array of one, two or three dimensions,
                         accessed through accessors.}];
   let parameters = (ins "::mlir::Type":$type,
-                        "unsigned":$dimension);
+                        "unsigned":$dimension,
+                        DefaultValuedParameter<"bool", "false">:$subbuffer);
   let assemblyFormat = 
-      "`<` `[` $dimension `,` $type `]` `>`";
+      "`<` `[` $dimension `,` $type (`,` $subbuffer^)? `]` `>`";
```

Comment:
Should this be a type property? What if we create a buffer from another one (i.e., a subbuffer) and then reuse that allocation to host a non-subbuffer?

Reply:
I think it makes sense to encode this in the type for now, as there are semantic differences. For example, you cannot create another sub-buffer from a buffer that is already a sub-buffer. Currently, we do not create allocations or values of this type, it only appears in `sycl.host.constructor`.

Reply:
Nice. LGTM in that case!